addons/account_edi/models/account_move.py

```diff
@@ -450,6 +454,34 @@ def _process_attachments_for_template_post(self, mail_template):
                 move_result.setdefault('attachments', []).extend(edi_attachments.get('attachments', []))
         return result
 
+    ####################################################
+    # Export Electronic Document
+    ####################################################
+
+    def action_download_electronic_invoice(self):
+        ids = ','.join(str(id) for id in self.filtered('edi_document_ids').ids)
+        if not ids:
```

Comment:
```suggestion
        if ids:
            return {...
```
This way you only return once :D